factor/totp/classes/factor.php

```diff
@@ -156,13 +153,11 @@ public function setup_factor_form_definition_after_data($mform) {
         $qrcode = $this->generate_qrcode($secret);
 
         $html = \html_writer::tag('p', $qrcode);
-        $xssallowedelements[] = $mform->addElement('static', 'scan', get_string('setupfactor:scan', 'factor_totp'), $html);
```

Comment:
Needs to be changed to only remove the `$xssallowedelements[] = ` section, otherwise we remove the form elements we want. Same on line 165 and 200

Reply:
grr - not sure what I was doing there - fixed that up.